src/Sylius/Bundle/CoreBundle/Processor/CatalogPromotionReprocessor.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Processor;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+class CatalogPromotionReprocessor implements CatalogPromotionReprocessorInterface
```

Comment:
This name does not resonate with me has anyone got better naming for it (also namespace is it ok?)

Reply:
How about `CatalogPromotionProcessor`?

Reply:
We have `CatalogPromotionProcessor` in this namespace - my class is just an extension for it cause it first clears promotions and then process them all again